modules/core/src/main/java/org/apache/ignite/internal/processors/cluster/GridClusterStateProcessor.java

```diff
@@ -924,14 +901,11 @@ protected IgniteCheckedException concurrentStateChangeError(ClusterState state,
                 throw new IgniteException("Node with BaselineTopology cannot join" +
                     " mixed cluster running in compatibility mode");
 
-            globalState = (DiscoveryDataClusterState) data.commonData();
+            globalState = (DiscoveryDataClusterState)data.commonData();
 
             compatibilityMode = true;
 
-            ctx.cache().context().readOnlyMode(readOnly(globalState.state()));
-
-            if (readOnly(globalState.state()))
-                ctx.cache().context().database().forceCheckpoint("Cluster read-only mode enabled");
+            ctx.cache().context().readOnlyMode(globalState.state() == ACTIVE_READ_ONLY);
```

Comment:
I've added force checkpoint by mistake. I thought that it could be helpful for the cluster backup in read-only mode. But, I forgot about rebalance and other inner activities. So, force checkpoint not needed there.